File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> map;

        for (auto& str : strs) {
            vector<int> count(26, 0);
            for (auto c : str) {
                count[c - 'a']++;
            }

            string key = "";

            for (auto i : count) {
                key = key + to_string(i) + ",";
            }

            map[key].push_back(str);
        }

        vector<vector<string>> ans;

        for (auto& pair : map) {
            ans.push_back(pair.second);

        }

        return ans;
        
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mp = {}
        for str in strs:
            count = [0] * 26
            for c in str:
                count[ord(c) - ord('a')] += 1
            
            key = tuple(count)
            
            # if key not in mp:
            #     mp[key] = []

            # mp[key].append(str)

            mp.setdefault(key, []).append(str)
           
        
        return list(mp.values())
        